### ISSUE TITLE
ensureDirectory function would be handy

### ISSUE THREAD

**Issue by tansaku** [perry-mitchell/webdav-client#118]
an ensureDirectory function that took a path and ensured that it existed would be pretty handy ... e.g.

```js
client.ensureDirectory('/tmp/foo/bar') // ensures that each directory in the path exists
```

**Comment by perry-mitchell** [comment]
This will be available via #253 by calling the following:

```typescript
client.createDirectory("/tmp/foo/bar", { recursive: true });
```

**Comment by eric-burel** [comment]
Hi, I suspect current implementation doesn't resist to race conditions. Between the moment you run "stats" and the moment your run "createDirectory", the directory might actually have been created. You should probably add an additional "try/catch" that ignores "405" errors when the creation fails for a subpath, it means that another call already created this subpath.
